src/QMCHamiltonians/SOECPComponent.h

```diff
@@ -47,12 +47,12 @@ class SOECPComponent : public QMCTraits
   std::vector<RadialPotentialType*> sopp_m;
 
 public:
-  SOECPComponent(){};
 
-  ///destructor
-  ~SOECPComponent(){};
-
-  SOECPComponent* makeClone(const ParticleSet& qp){};
+  SOECPComponent* makeClone(const ParticleSet& qp)
+  {
+    APP_ABORT("SOECPComponent::makeClone not yet implemented");
```

Comment:
Instead of APP_ABORT I've started throwing std::runtime_error in these cases. One big plus is this gives us a path to better error handling when concurrency is non trivial and you can check for the the throw in a catch test program.